src/domain/app/header/Header.tsx

```diff
@@ -1,26 +1,125 @@
+      label: t(`common:languages:${language}`),
+      value: language,
+    }));
+  }, [t]);
+
+  const changeLanguage = (newLanguage: OptionType) => (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
+  ) => {
+    const url = window.location.href;
+    event.preventDefault();
+    router.push(updateLocaleParam(url, locale, newLanguage.value));
+  };
+
+  const isTabActive = (pathname: string): boolean => {
+    return (
+      typeof window !== 'undefined' &&
+      window.location.pathname.startsWith(pathname)
+    );
+  };
+
+  const goToPage = (pathname: string) => (
+    event?: React.MouseEvent<HTMLAnchorElement>
+  ) => {
+    event?.preventDefault();
+    router.push(pathname);
+    closeMenu();
+  };
+
+  const logoLang = locale === 'sv' ? 'sv' : 'fi';
+
+  const navigationItems = [
+    {
+      label: t('header:navigation:about'),
+      url: `/${locale}${ROUTES.ABOUT}`,
+    },
+    {
+      label: t('header:navigation:culturalEducation'),
+      url: `/${locale}${ROUTES.CULTURAL_EDUCATION}`,
+    },
+    {
+      label: t('header:navigation:helsinkiMoving'),
+      url: `/${locale}${ROUTES.MOVING}`,
+    },
+    {
+      label: t('header:navigation:now'),
+      url: `/${locale}${ROUTES.NOW}`,
+    },
+    {
+      label: t('header:navigation:forOrganizer'),
+      url: `/${locale}${ROUTES.FOR_ORGANIZER}`,
+      icon: <IconSignin />,
+    },
+  ];
 
   return (
-    <header className={styles.header}>
-      <a
-        className={styles.skipToContent}
-        href={`#${MAIN_CONTENT_ID}`}
-        aria-label={t('common:linkSkipToContent')}
-      >
-        {t('common:linkSkipToContent')}
-      </a>
-      <MobileNavbar className={styles.mobileNavbar} />
-      <Navbar className={styles.navbar} />
-    </header>
+    <Navigation
+      menuOpen={menuOpen}
+      onMenuToggle={toggleMenu}
+      menuToggleAriaLabel={t('header:menuToggleAriaLabel')}
+      skipTo={`#${MAIN_CONTENT_ID}`}
+      skipToContentLabel={t('common:linkSkipToContent')}
+      className={styles.navigation}
+      onTitleClick={goToPage(`/${locale}${ROUTES.HOME}`)}
+      logoLanguage={logoLang}
+      title={t('common:appName')}
+    >
+      <Navigation.Row variant="inline">
+        {navigationItems.map((item, index) => (
+          <Navigation.Item
+            key={index}
+            active={isTabActive(item.url)}
+            className={styles.navigationItem}
+            href={item.url}
+            label={item.label}
+            onClick={goToPage(item.url)}
+            icon={item.icon}
+          />
+        ))}
+      </Navigation.Row>
+      <Navigation.Actions>
+        <Navigation.LanguageSelector
+          buttonAriaLabel={t('header:changeLanguage')}
+          className={styles.languageSelector}
+          label={t(`common:languages:${locale}`)}
+          icon={<IconGlobe />}
+        >
+          {languageOptions.map((option) => (
+            <Navigation.Item
+              key={option.value}
+              href="#"
```

Comment:
Enter key not working  in these items, maybe all of them need the specific url

Reply:
actually its a bug of hds component i think. its changing the language, but not closing the menu even on click.

Reply:
can you retest? i think with updated hds it works now